plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -706,13 +706,142 @@ export class PersonOverrideWriter {
+
+        const { rows: transitiveUpdates } = await this.postgres.query(
+            tx,
+            SQL`
+                UPDATE
+                    posthog_flatpersonoverride
+                SET
+                    override_person_id = ${overrideDetails.override_person_id},
+                    version = COALESCE(version, 0)::numeric + 1
+                WHERE
+                    team_id = ${overrideDetails.team_id} AND override_person_id = ${overrideDetails.old_person_id}
+                RETURNING
+                    old_person_id,
+                    version,
+                    oldest_event
+            `,
+            undefined,
+            'transitivePersonOverrides'
+        )
+
+        status.debug('🔁', 'person_overrides_updated', { transitiveUpdates })
+
+        const personOverrideMessages: ProducerRecord[] = [
+            {
+                topic: KAFKA_PERSON_OVERRIDE,
+                messages: [
+                    {
+                        value: JSON.stringify({
+                            team_id: overrideDetails.team_id,
+                            old_person_id: overrideDetails.old_person_id,
+                            override_person_id: overrideDetails.override_person_id,
+                            oldest_event: castTimestampOrNow(overrideDetails.oldest_event, TimestampFormat.ClickHouse),
+                            merged_at: castTimestampOrNow(mergedAt, TimestampFormat.ClickHouse),
+                            version: 0,
+                        }),
+                    },
+                    ...transitiveUpdates.map(({ old_person_id, version, oldest_event }) => ({
+                        value: JSON.stringify({
+                            team_id: overrideDetails.team_id,
+                            old_person_id: old_person_id,
+                            override_person_id: overrideDetails.override_person_id,
+                            oldest_event: castTimestampOrNow(oldest_event, TimestampFormat.ClickHouse),
+                            merged_at: castTimestampOrNow(mergedAt, TimestampFormat.ClickHouse),
+                            version: version,
+                        }),
+                    })),
+                ],
+            },
+        ]
+
+        return personOverrideMessages
+    }
+
+    public async getPersonOverrides(teamId: number): Promise<PersonOverrideDetails[]> {
+        const { rows } = await this.postgres.query(
+            PostgresUse.COMMON_WRITE,
+            SQL`
+                SELECT
+                    team_id,
+                    old_person_id,
+                    override_person_id,
+                    oldest_event
+                FROM posthog_flatpersonoverride
+                WHERE team_id = ${teamId}
+            `,
+            undefined,
+            'getPersonOverrides'
+        )
+        return rows.map((row) => ({
+            ...row,
+            team_id: parseInt(row.team_id), // XXX: pg returns bigint as str (reasonably so)
```

Comment:
This is not a good idea in general terms, but like the other `getPersonOverrides` above, this is only intended to be used in tests.

On second thought, there's probably no harm in replacing `parseInt` with `BigInt`, though? It seems like it should be safe…?

```node
> BigInt('10') == 10
true
```

Reply:
Yeah, seems harmless (either way?).

Reply:
Hm, Jest doesn't like it after all,

```diff
diff --git a/plugin-server/src/worker/ingestion/person-state.ts b/plugin-server/src/worker/ingestion/person-state.ts
index f49ee1ad33..e33fcad987 100644
--- a/plugin-server/src/worker/ingestion/person-state.ts
+++ b/plugin-server/src/worker/ingestion/person-state.ts
@@ -832,7 +832,7 @@ export class FlatPersonOverrideWriter {
         )
         return rows.map((row) => ({
             ...row,
-            team_id: parseInt(row.team_id), // XXX: pg returns bigint as str (reasonably so)
+            team_id: BigInt(row.team_id), // pg returns bigint as str
             oldest_event: DateTime.fromISO(row.oldest_event),
         }))
     }
```

causes this error:

```
  ● person overrides writer: flat › handles direct overrides

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

      Array [
        Object {
          "old_person_id": "018c6f1a-36df-0000-4ee6-e4ef7823a3f8",
          "oldest_event": "1969-12-31T16:00:00.000-08:00",
          "override_person_id": "018c6f1a-36df-0001-a5b9-21531ca2319e",
    -     "team_id": 474353818,
    +     "team_id": 474353818n,
        },
      ]

      2152 |         })
      2153 |
    > 2154 |         expect(await writer.getPersonOverrides(teamId)).toEqual([{ ...defaults, ...override }])
           |                                                         ^
      2155 |     })
      2156 |
      2157 |     it('handles transitive overrides', async () => {

      at Object.toEqual (tests/worker/ingestion/person-state.test.ts:2154:57)
```

Not sure why, as the values compare just fine:

```
> 486884794 == 486884794n
true
```

Looks like that equality test comes from somewhere in here, so it's not using the equality operator directly: https://github.com/jestjs/jest/blob/e54c0ebb048e10331345dbe99f8ec07654a43f1c/packages/expect-utils/src/jasmineUtils.ts#L63-L212

`teamId` is used all over in these tests so it seems like a whole different can of worms to try and change it to `BigInt` consistently, so going to avoid falling down that rabbit hole and leave it as-is, I guess. (`posthog_team.id` isn't a `bigint` on the table schema, either, FWIW -- at least as it exists in the repository, I didn't check the production schema.)

Reply:
> `posthog_team.id` isn't a bigint on the table schema, either, FWIW

Oh, weird. To be honesty when I read your comment I was 🤨 that we're planning for >2billion teams already. Not that it isn't nice to be prepared, but this seems pretty low priority to think about, lol.

Reply:
Yeah, I just grabbed it from `PersonOverrideMapping` myself. I just assumed there was a "always use bigserial for primary keys" to avoid surprises later.

Knowing that, maybe it makes sense to drop these back down to a plain old 4 byte integer for consistency with the actual team primary key? Though these tables should ideally stay pretty small through overrides, so maybe it doesn't matter much at all? 🤷‍♂️ 

I do like the idea of 2 billion teams though.